taskchampion/src/replica.rs

```diff
@@ -759,6 +771,37 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn commit_reversed_operations() -> Result<()> {
+        let uuid1 = Uuid::new_v4();
+        let uuid2 = Uuid::new_v4();
+        let uuid3 = Uuid::new_v4();
+
+        let mut rep = Replica::new_inmemory();
+
+        let mut ops = Operations::new();
+        ops.push(Operation::UndoPoint);
+        rep.create_task(uuid1, &mut ops).unwrap();
+        ops.push(Operation::UndoPoint);
+        rep.create_task(uuid2, &mut ops).unwrap();
+        rep.commit_operations(dbg!(ops))?;
+        assert_eq!(rep.num_undo_points().unwrap(), 2);
+
+        // Trying to reverse-commit the wrong operations fails.
+        let ops = vec![Operation::Delete {
+            uuid: uuid3,
+            old_task: TaskMap::new(),
+        }];
+        assert!(!rep.commit_reversed_operations(ops)?);
+
+        // Commiting the correct operations succeeds
+        let ops = rep.get_undo_operations()?;
+        assert!(rep.commit_reversed_operations(ops)?);
+        assert_eq!(rep.num_undo_points().unwrap(), 1);
+
+        Ok(())
+    }
+
```

Comment:
In the spirit of fairness, this *does* seem like a valuable mutation to test against:

> taskchampion/src/replica.rs:355:12: delete ! in Replica::commit_reversed_operations

And a valuable test. :+1: